skills/src/main/java/cm/aptoide/skills/SkillsViewModel.kt

```diff
@@ -53,10 +57,19 @@ class SkillsViewModel @Inject constructor(
     const val RESULT_INVALID_URL = 7
     const val RESULT_INVALID_USERNAME = 8
     const val RESULT_ROOT_ERROR = 9
+    const val RESULT_WALLET_VERSION_ERROR = 10
     const val GET_ROOM_RETRY_MILLIS = 3000L
     const val AUTHENTICATION_REQUEST_CODE = 33
   }
 
+  fun updateClick(): Intent {
+    val appPackageName: String = packageName
+    return Intent(
+      Intent.ACTION_VIEW,
+      Uri.parse("market://details?id=$appPackageName")
+    )
+  }
```

Comment:
The ViewModel should not have logic, this should be a useCase and invoked in the viewModel to seperate the layers
Especially because the ViewModel should not receive a String thats only for this use case.

Ideally this logic would be reused from the logic we have in the :app module but its not possible currently because of the way we have the module depencies.

The useCase would look loke the BuildUpdateIntentUseCase() in the :app module

Reply:
changed to use case